Helper_Scripts/Agent-Tools/Audit-Agent/audit_agent.sh

```
@@ -0,0 +1,202 @@
+run_project_checks() {
+    local file_path="$1"
+    local project_root="$(git rev-parse --show-toplevel 2>/dev/null || pwd)"
+    
+    # Run linting if available
+    if [[ -f "$project_root/package.json" ]]; then
+        # Check if file is JavaScript/TypeScript
+        if [[ "$file_path" =~ \.(js|ts|jsx|tsx)$ ]]; then
+            if command -v npm >/dev/null 2>&1; then
+                # Try to run eslint if available
+                if npm list eslint >/dev/null 2>&1; then
+                    cd "$project_root"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)